### ISSUE TITLE
Getting `non-matching-header` instead of `missing-header` error

### ISSUE THREAD

**Issue by delenamalan** [frictionlessdata/goodtables-py#294]
When a dataset contains a subset of the headers defined in the schema, I would expect to get one or more `missing-header` (defined as "The data doesn't contain a header defined in the schema.") errors, but currently I get `non-matching-header`("The header's name in the schema is different from what's in the data.") errors.

For example, if my schema contains the fields `id`, `last_name` and `first_name` and my dataset contains only the `id` and `last_name` headers, I get the following error:
`{'code': 'non-matching-header', 'column-number': 3, 'message': 'Header in column 3 doesn\'t match field name "first_name" in the schema', 'message-data': {'field_name': 'first_name'}}]}`

This happens independent of whether `order_fields` is not true or not. 

## How to reproduce
Add a new test scenario called `missing_header.yml` to `tests/scenarios` with the following content and run `py.test`:
```
primary_key_field:
  source:
    - [id, last_name]
    - [1, Alex]
    - [2, John]
    - [3, Paul]
  schema:
    fields:
      - {name: id, type: number}
      - {name: last_name}
      - {name: first_name}
    primaryKey: id
  checks:
    - schema
    - structure
  report:
    - [1, None, 3, 'missing-header']

```

**Comment by roll** [comment]
@delenamalan 
Thanks for reporting!

**Comment by roll** [comment]
Fixed in #298
